### ISSUE TITLE
Buffer Underflow Exception

### ISSUE THREAD

**Issue by SoufianeOuzzedi** [NewtronLabs/EasyBluetooth#6]
[Android Studio 4.1]
When running the slightly modified sample code of the plugin, I get this error as soon as the two devices connect to each other. Here's the code of my main activity and the zip containing the entire Android Studio project : 

```
public class MainActivity extends AppCompatActivity {

    private String selectedServerAddress;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        FillPairedDevicesSpinner();

        findViewById(R.id.startServerButton).setOnClickListener(onClickListener);
        findViewById(R.id.connectToServerButton).setOnClickListener(onClickListener);
    }

    private View.OnClickListener onClickListener = new View.OnClickListener() {
        @Override
        public void onClick(final View v) {
            switch(v.getId()){
                case R.id.startServerButton:
                    LaunchServer();
                    break;
                case R.id.connectToServerButton:
                    LaunchClient();
                    break;
                default:break;
            }
        }
    };


    private void FillPairedDevicesSpinner()
    {
        Log.d("easyBt", "Filling spinner content...");

        //Fetch list of paired devices names
        Set<BluetoothDevice> pairedDevicesSet = BluetoothAdapter.getDefaultAdapter().getBondedDevices();
        ArrayList<String> pairedDevicesStrings = new ArrayList<String>();
        Log.d("easyBt", "Found "+pairedDevicesSet.size()+" devices paired");
        for(BluetoothDevice device : pairedDevicesSet)
        {
            Log.d("easyBt", "-> filling spinner with : "+device.getName()+" -- "+device.getAddress());
            pairedDevicesStrings.add(device.getName()+ "-@-" + device.getAddress());
        }

        //Set up spinner content
        Spinner dynamicSpinner = (Spinner) findViewById(R.id.pairedDevicesSpinner);

        String[] items = new String[pairedDevicesStrings.size()];
        ArrayAdapter<String> adapter = new ArrayAdapter<String>(this,
                android.R.layout.simple_spinner_item, pairedDevicesStrings);

        dynamicSpinner.setAdapter(adapter);
        dynamicSpinner.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
            @Override
            public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
                String choice = (String) parent.getItemAtPosition(position);
                selectedServerAddress = choice.split("-@-")[1];
                Log.d("easyBt", "Selected Item : "+ choice + " which correspond to address : "+ selectedServerAddress);
            }

            @Override
            public void onNothingSelected(AdapterView<?> parent) {
                // TODO Auto-generated method stub
            }
        });
    }


    private void LaunchServer() {
        Log.d("easyBt", "Clicked to start server...");

        // Define your service and it's UUID
        IBluetoothServer btServer = new BluetoothServer.Builder(this.getApplicationContext(),
                "EasyBtService",ParcelUuid.fromString("00001101-0000-1000-8000-00805F9B34FB"))
                .build();

        if(btServer == null)
        {
            // Server could not be created.
        }
        else
        {
            // Block until a client connects.
            IBluetoothClient btClient = btServer.accept();
            // Set a data callback to receive data from the remote device.
            btClient.setDataCallback(new DataCallback());
            // Set a connection callback to be notified of connection changes.
            btClient.setConnectionCallback(new ConnectionCallback());
            // Set a data send callback to be notified when data is sent of fails to send.
            btClient.setDataSentCallback(new DataSentCallback());
            btClient.sendData("ServerGreeting", "Hello Client".getBytes());
            //We don't want to accept any other clients.
            btServer.disconnect();
        }
    }

    private void LaunchClient() {
        Log.d("easyBt", "Clicked to connect client...");

        // Find the server Bluetooth device.
        BluetoothDevice serverDev = BluetoothAdapter.getDefaultAdapter().getRemoteDevice(selectedServerAddress);
        IBluetoothClient client = new BluetoothClient.Builder(this.getApplication(), serverDev, ParcelUuid.fromString("00001101-0000-1000-8000-00805F9B34FB"))
                // We want to be notified when connection completes.
                .setConnectionCallback(new ConnectionCallback())
                // Let's also get notified if it fails
                .setConnectionFailedListener(new ConnectionFailedListener())
                // Receive data from the server
                .setDataCallback(new DataCallback())
                // Be notified when the data is sent to the server or fails to send.
                .setDataSentCallback(new DataSentCallback())
                .build();

        // Connect to server
        client.connect();
    }
}
```

[NativeCompagnonBluetooth.zip](https://github.com/NewtronLabs/EasyBluetooth/files/5872102/NativeCompagnonBluetooth.zip)

**Comment by NewtronLabs** [comment]
Hi @SoufianeOuzzedi,

Thank you for the detailed question. We will get back to you shortly.

Thanks,
Amy

**Comment by NewtronLabs** [comment]
Hi @SoufianeOuzzedi!

The issue has been fixed in version `4.0.2`. You can go ahead and upgrade to that version.

Let us know if you notice another issue, we are always happy to lend a helping hand.

Thanks,
Amy
